src/naive/date.rs

```diff
@@ -552,12 +552,24 @@ impl NaiveDate {
     ///            Ok(NaiveDate::from_ymd_opt(2014, 5, 17).unwrap()));
     /// ```
     ///
-    /// Out-of-bound dates or insufficient fields are errors.
+    /// If the format string doesn't contain a day field, it defaults to the first day.
+    /// If year and month are given it defaults to the first day of the month.
+    /// If year and week are given it defaults to the first day of the week.
+    ///
```

Comment:
I really like this detailed explanation in the docstring.

However, it's somewhat confusing.

My suggestion includes referring specifically to `fmt`, use the phrase _specifier_ (that is used elsewhere), and being more explicit about the inferred subject:

    /// If `fmt` does not contain a day specifier then the day value defaults to the first day of the week.
    /// If `fmt` only has year and month specifiers then the day value defaults to the first day of the month.
    /// If `fmt` only has year and week specifier then the day value value defaults to the first day of the week.

(I'm not sure if my proposed sentences are technically correct, you'll have to review that)

Reply:
Thank you for the suggestion. I have updated it to:
```rust
    /// If `fmt` does not contain enough specifiers to get a complete date, parsing inserts a
    /// default value:
    /// - If `fmt` has year and month specifiers parsing assumes it is the first day of the month.
    /// - If `fmt` has year and week specifiers parsing assumes it is the first day of the week.
```